python/cisa_risk_meter/build_cisa_risk_meter.py

```diff
@@ -0,0 +1,334 @@
+import os
+import sys
+import time
+import json
+import logging
+import requests
+
+# Humble constants.
+API_MAX_PAGES = 20
+CISA_CUSTOM_FIELD_NAME = "CISA"
+CISA_RISK_METER_NAME = "CISA Exploited Vulnerabilities"
+VULN_UPDATE_LIMIT = 5000
+
+# Maximum page size
+SEARCH_PAGE_SIZE = 5000
+
+# This might have to changed depending on your setup.
+KENNA_BASE_URL = "https://api.kennasecurity.com/"
+
+# Dump JSON in a pretty format.  Great for debugging.
+def print_json(json_obj):
+    print(json.dumps(json_obj, sort_keys=True, indent=2))
+
+# Get the CISA catalog.
+def get_cisa_catalog():
+    get_cisa_catalog_url = f"https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json"
+    headers = { 'Accept': 'application/json',
+                'Content-Type': 'application/json; charset=utf-8' }
+
+    response = requests.get(get_cisa_catalog_url, headers=headers)
+    if response.status_code != 200:
+        print(f"List CISA Catalog Error: {response.status_code} with {get_cisa_catalog_url}")
+        logging.error(f"List CISA Catalog API, {get_cisa_catalog_url} error: {resp_json}")
+        sys.exit(1)
+
+    resp_json = response.json()
+
+    # Return the CISA catalog.
+    return resp_json
+
+# Checks for a custom_field name is associated with the vulnerability.
+# Returns the custom field or None.
+def get_custom_field(vuln, custom_field_name):
+    custom_fields = vuln['custom_fields']
+
+    for custom_field in custom_fields:
+        if custom_field['name'] == custom_field_name:
+            return custom_field
```

Comment:
Do we only care for the first custom_field that has the name match if there are more one matches?

Reply:
We only need the first one to get the custom field ID.  I will add this to the comments.